cabal-install/Distribution/Client/Sandbox/Index.hs

```diff
@@ -156,39 +158,56 @@ addBuildTreeRefs verbosity path l' refType = do
     updatePackageIndexCacheFile verbosity path
       (path `replaceExtension` "cache")
 
+data DeleteSourceError = ErrNonregisteredSource { nrPath :: FilePath }
+                       | ErrNonexistentSource   { nePath :: FilePath } deriving Show
+
 -- | Remove given local build tree references from the index.
 --
 -- Returns a tuple consisting of removed build tree refs and mappings from
 -- provided build tree refs to corresponding full directory paths).
 removeBuildTreeRefs :: Verbosity -> FilePath -> [FilePath]
-                       -> IO ([FilePath], [(FilePath, FilePath)])
+                       -> IO ([Either DeleteSourceError FilePath],
+                              (FilePath -> FilePath)) -- ^ A tuple consisting of:
```

Comment:
I changed the Haddock string for this function, so this comment is no longer needed.